lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -15,16 +16,71 @@ class FirebaseAppDistributionAction < Action
       FIREBASECMD_ACTION = "appdistribution:distribute".freeze
       BASE_URL = "https://firebaseappdistribution.googleapis.com"
       PATH = "/v1alpha/apps/"
+      MAX_POLLING_RETRIES = 60
+      POLLING_INTERVAL_S = 2
 
       extend Helper::FirebaseAppDistributionHelper
 
       def self.run(params)
         params.values # to validate all inputs before looking for the ipa/apk
-        validate_app!(params[:app])
+        app_id = nil
+        platform = Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+        binary_path = Shellwords.escape(params[:ipa_path] || params[:apk_path])
+
+        if params[:app] # Set app_id if it is specified as a parameter
+          app_id = params[:app]
+        else
+          if platform == :ios
+            archive_path = Actions.lane_context[SharedValues::XCODEBUILD_ARCHIVE]
+            if archive_patharchive_path
+              app_id = findout_ios_app_id_from_archive(archive_path)
+            end
+          end
+          if app_id.nil?
+            app_id = UI.input("Your app's Firebase App ID. You can find the App ID in the Firebase console, on the General Settings page: ")
+          end
+        end
+
+        upload_token = get_upload_token(app_id, binary_path)
+        MAX_POLLING_RETRIES.times do
+          status = upload_status(upload_token, app_id)
+          if status == "SUCCESS"
+            UI.message("App Uploaded Successfully!")
+            break
+          elsif status == "IN_PROGRESS"
+            sleep(POLLING_INTERVAL_S)
+          else
+            UI.message("Uploading APK")
```

Comment:
If the binary has not been uploaded before, let's show the same two messages that the gradle plugin shows i.e.
```
UI.message("This APK has not been uploaded before.")
UI.message("Uploading the APK.")
```